ci/scripts/run-backfill-tests.sh

```diff
@@ -76,12 +353,89 @@ test_backfill_tombstone() {
 
   ./risedev psql -c "CREATE MATERIALIZED VIEW m1 as select * from tomb;"
   echo "--- Kill cluster"
-  cargo make kill
+  kill_cluster
+  wait
+}
+
+test_backfill_restart_cn_recovery() {
+   echo "--- e2e, $CLUSTER_PROFILE, test_background_restart_cn_recovery"
+   cargo make ci-start $CLUSTER_PROFILE
+
+   # Test before recovery
+   sqllogictest -d dev -h localhost -p 4566 "$COMMON_DIR/create_table.slt"
+   sqllogictest -d dev -h localhost -p 4566 "$COMMON_DIR/create_bg_mv.slt"
+   sleep 1
+   OLD_PROGRESS=$(run_sql "SHOW JOBS;" | grep -E -o "[0-9]{1,2}\.[0-9]{1,2}")
+
+   # Restart 1 CN
+   restart_cn
+
+   # Give some time to recover.
+   sleep 3
+
+   # Test after recovery
+   sqllogictest -d dev -h localhost -p 4566 "$COMMON_DIR/validate_one_job.slt"
+
+   # Recover the mview progress
+   sleep 5
+
+   NEW_PROGRESS=$(run_sql "SHOW JOBS;" | grep -E -o "[0-9]{1,2}\.[0-9]{1,2}")
+
+   if [[ ${OLD_PROGRESS%.*} -le ${NEW_PROGRESS%.*} ]]; then
+     echo "OK: $OLD_PROGRESS smaller or equal to $NEW_PROGRESS"
+   else
+     echo "FAILED: $OLD_PROGRESS larger than $NEW_PROGRESS"
+     exit 1
+   fi
+
+   # Trigger a bootstrap recovery
+   pkill compute-node
+   kill_cluster
+   rename_logs_with_prefix "before-restart"
+   sleep 10
+   cargo make dev $CLUSTER_PROFILE
+
+   # Recover mview progress
+   sleep 5
+
+   OLD_PROGRESS=$NEW_PROGRESS
+   NEW_PROGRESS=$(run_sql "SHOW JOBS;" | grep -E -o "[0-9]{1,2}\.[0-9]{1,2}")
+
+   if [[ ${OLD_PROGRESS%.*} -le ${NEW_PROGRESS%.*} ]]; then
+     echo "OK: $OLD_PROGRESS smaller or equal to $NEW_PROGRESS"
+   else
+     echo "FAILED: $OLD_PROGRESS larger than $NEW_PROGRESS"
+     exit 1
+   fi
+
+   sleep 60
+
+   # Test after backfill finished
+   sqllogictest -d dev -h localhost -p 4566 "$COMMON_DIR/validate_backfilled_mv.slt"
```

Comment:
I will do some analysis for https://github.com/risingwavelabs/risingwave/pull/12167#discussion_r1361676140 first.

It's actually why I wrote tests these way, since progress did not show when running in deterministic simulation.

Still figuring out root cause.

Reply:
Agreed with @BugenZhao , it's too hard to maintain and reproduce it if the test fails. Besides we'd better not rely on progress and sleep to validate this feature, it's too flaky as said. +1 for utilizing the simulation integration tests for it like what scaling test does.

Reply:
See https://github.com/risingwavelabs/risingwave/issues/12957